### PR TITLE
bump versions and add scalafmt

### PR DESCRIPTION
**Pull request by russwyte**
Used default format with align which makes looking at pattern matches very easy. 

scalafmt will make it a lot easier to contribute. 

See: https://scalameta.org/scalafmt/docs/installation.html

I also bumped scala/scala.js/sbt versions to current versions.

### CODE REVIEW

File: src/main/scala/com/github/ahnfelt/react4s/Css.scala [hunk 1]
```diff
@@ -3,109 +3,140 @@ package com.github.ahnfelt.react4s
...
 object Css {
 
-    def selector(selector : String, children : CssChild*) = CssSelector(selector, children)
-    def media(query : String, children : CssChild*) = CssMediaQuery(query, children)
-    def pseudo(pseudoClass : String, children : CssChild*) = CssPseudoClass(pseudoClass, children)
-    def cssClass(className : String, children : CssChild*) = CssSelector("." + className, children)
-    def directChild(tagName : String, children : CssChild*) = CssSelector(">" + tagName, children)
-    def mediaSmall(children : CssChild*) = CssMediaQuery("(max-width: 991px)", children)
-    def mediaLarge(children : CssChild*) = CssMediaQuery("(min-width: 992px)", children)
+  def selector(selector: String, children: CssChild*) =
```
Comment (zakpatterson) [inline src/main/scala/com/github/ahnfelt/react4s/Css.scala]: Why did scalafmt break some of these lines?

Comment (russwyte) [reply src/main/scala/com/github/ahnfelt/react4s/Css.scala]: I am not sure - it must be part of the default rules.

Comment (russwyte) [reply src/main/scala/com/github/ahnfelt/react4s/Css.scala]: Ah I had the config file in the wrong spot. Pushed a change - see if you like it better.

Comment (russwyte) [reply src/main/scala/com/github/ahnfelt/react4s/Css.scala]: We can tweak it as much as we like:
https://scalameta.org/scalafmt/docs/configuration.html

**Comment by Ahnfelt** [comment]
I really appreciate your contribution, but this stylistic change is proving more laborious for me than I anticipated. Every time a change is made, I have to check essentially the whole code base all over from scratch. I think our energy is better spent elsewhere in the codebase, at least for the time being.

**Pull request closed without merging**